python/lsst/analysis/tools/actions/plot/matrixPlot.py

```diff
@@ -0,0 +1,384 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("MatrixPlot",)
+
+from typing import TYPE_CHECKING, Any, Mapping
+
+import astropy.visualization as apViz
+import matplotlib.patheffects as mpl_path_effects
+import matplotlib.pyplot as plt
+import numpy as np
+from astropy.visualization.mpl_normalize import ImageNormalize
+from lsst.pex.config import Config, ConfigDictField, DictField, Field, ListField
+
+from ...interfaces import PlotAction, Vector
+from .plotUtils import addPlotInfo
+
+if TYPE_CHECKING:
+    from matplotlib.figure import Figure
+
+    from ...interfaces import KeyedData, KeyedDataSchema
+
+
+class guideLinesConfig(Config):
+    lines = DictField[float, str](
+        doc=("Dictionary of x/y-values and the labels where vertical/horizontal lines are drawn."),
+        optional=False,
+    )
+
+    color = Field[str](
+        doc="The color of the lines and labels.",
+        default="red",
+    )
+
+    outlineColor = Field[str](
+        doc="The color of the outline around the lines and labels.",
+        default="white",
+    )
+
+    linestyle = Field[str](
+        doc="The style of the lines.",
+        default="--",
+    )
+
+
+class MatrixPlot(PlotAction):
+    """Make the plot of a matrix (2D array)."""
+
+    matrixKey = Field[str](
+        doc="The key for the input matrix.",
+        default="matrix",
+    )
+
+    xAxisLabel = Field[str](
+        doc="The label to use for the x-axis.",
+        default="",
+    )
+
+    yAxisLabel = Field[str](
+        doc="The label to use for the y-axis.",
+        default="",
+    )
+
+    colorbarLabel = Field[str](
+        doc="The label to use for the colorbar.",
+        default="",
+    )
+
+    vmin = Field[float](
+        doc="The vmin value for the colorbar.",
+        default=None,
+    )
+
+    vmax = Field[float](
+        doc="The vmax value for the colorbar.",
+        default=None,
+    )
+
+    figsize = ListField[float](
+        doc="The size of the figure.",
+        default=[5, 5],
+        maxLength=2,
+    )
+
+    title = Field[str](
+        doc="The title of the figure.",
+        default="",
+    )
+
+    titleFontSize = Field[int](
+        doc="The font size for the title.",
+        default=12,
+    )
+
+    xAxisTickValues = ListField[float](
+        doc="List of x-axis tick values. If not set, the ticks will be set automatically by matplotlib.",
+        default=None,
+    )
+
+    xAxisTickLabels = DictField[float, str](
+        doc="Dictionary of x-axis tick positions and labels. These can work independent of "
```

Comment:
These are applied as minor ticks, right? Maybe add a longer note about that in the class docstring, especially if it's possible to misconfigure them such that the major and minor tick labels overlap.